### ISSUE TITLE
TODO: Investigate original Scala version of RemoteDaemon.GetChild, and create proper implementation of it

### ISSUE THREAD

**Issue by legezam** [akkadotnet/akka.net#360]
Source: /src/core/Akka.Remote/RemoteDaemon.cs#155
```C#
        /// <summary>
        ///     Gets the child.
        /// </summary>
        /// <param name="name">The name.</param>
        /// <returns>ActorRef.</returns>
        public override ActorRef GetChild(IEnumerable<string> name)
        {
            string[] parts = name.ToArray();
            //TODO: I have no clue what the scala version does
            if (!parts.Any())
                return this;

            string n = parts.First();
            if (string.IsNullOrEmpty(n))
                return this;

            for (int i = parts.Length; i >= 0; i--)
            {
                string joined = string.Join("/", parts, 0, i);
                InternalActorRef child;
                if (TryGetChild(joined, out child))
                {
                    //longest match found
                    IEnumerable<string> rest = parts.Skip(i);
                    return child.GetChild(rest);
                }
            }
            return Nobody;
        }
```

**Comment by maxim-s** [comment]
PR #1610 relate to #1544

**Comment by rogeralsing** [comment]
Have we verified if #1610 fixes #1544?

**Comment by maxim-s** [comment]
i wrote my comment ,
https://github.com/akkadotnet/akka.net/issues/1544#issuecomment-166070834
looks like it's independent issues
5 янв. 2016 г. 3:06 PM пользователь "Roger Johansson" <
notifications@github.com> написал:

**Comment by sean-gilliam** [comment]
Got to love that xmldoc comment.

**Comment by Horusiath** [comment]
@Aaronontheweb @maxim-s what is the state of this issue?

**Comment by maxim-s** [comment]
@Aaronontheweb @Horusiath Updated PR #1610

**Comment by Silv3rcircl3** [comment]
Fixed via #1610
